src/App.tsx

```diff
@@ -1,25 +1,31 @@
-import React from "react";
+import React, { lazy, Suspense } from "react";
 import {
   BrowserRouter as Router,
   Route,
   Routes,
   useParams,
 } from "react-router-dom";
-import Home from "./pages/Home";
-import Calendar from "./pages/Calendar";
-import NotFound from "./pages/NotFound";
-import Register from "./pages/Register";
-import Resources from "./pages/Resources";
+
+// Lazy load pages
+const Home = lazy(() => import("./pages/Home"));
+const Calendar = lazy(() => import("./pages/Calendar"));
+const NotFound = lazy(() => import("./pages/NotFound"));
+const Register = lazy(() => import("./pages/Register"));
+const Resources = lazy(() => import("./pages/Resources"));
+const Members = lazy(() => import("./pages/Members"));
+const MemberProfile = lazy(() => import("./pages/MemberProfile"));
 
 const App: React.FC = () => {
   return (
     <Router>
-      <Routes>
-        <Route path="/" element={<Home />} />
-        <Route path="/register" element={<Register />} />
-        <Route path="/resources" element={<Resources />} />
-        <Route path="/:year/*" element={<YearRoutes />} />
-      </Routes>
+      <Suspense fallback={<div>Loading...</div>}>
```

Comment:
use spinner or better UI approaches instead of just loading. aile lai can be considered